core/src/main/java/greencity/config/SecurityConfig.java

```diff
@@ -121,6 +121,7 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
                                 "/user/activatedUsersAmount",
                                 "/user/{userId}/habit/assign",
                                 "/token",
+                                "/user/findByEmail",
```

Comment:
Why have you given access to this end-point to all users, even if it's unauthorized?